src/include/serial_inline.h

```diff
@@ -300,32 +299,12 @@ __wt_update_serial(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_PAGE *page
         return (0);
 
     /*
-     * We would like to call __wt_txn_update_oldest only in the event that there are further updates
-     * to this page, the check against WT_TXN_NONE is used as an indicator of there being further
-     * updates on this page.
+     * Look for obsolete updates if the page size is more than 50% of the maximum split page size
+     * allowed by btree. This is to prevent unnecessary page splits due to the obsolete updates
+     * present on the page.
      */
-    if ((txn = page->modify->obsolete_check_txn) != WT_TXN_NONE) {
-        obsolete_timestamp = page->modify->obsolete_check_timestamp;
-        if (!__wt_txn_visible_all(session, txn, obsolete_timestamp)) {
```

Comment:
I'm not convinced this is always an unnecessary check. What if a page grows larger than 50% due to new records being inserted, there still isn't any reason to aggressively do obsolete update checking.

Reply:
The reasons I feel this code is not doing much.
1. The `pinned timestamp` is used as `obsolete_check_timestamp`. This timestamp is already globally visible. So setting it or not setting doesn't do anything.
2. The `obsolete_check_txn` is set to the `last_running` transaction in the system. In most of the scenarios, by the time we check for an obsolete update again, this transaction may have finished already, and the `oldest_id` may have moved beyond it. 

I am fine to add back this change, but only for the transaction ID, as the timestamp check is not required.